flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/AbstractIcebergEnumerator.java

```diff
@@ -93,6 +98,28 @@ public void handleSourceEvent(int subtaskId, SourceEvent sourceEvent) {
     }
   }
 
+  @Override
+  public void handleSourceEvent(int subTaskId, int attemptNumber, SourceEvent sourceEvent) {
+    if (sourceEvent instanceof SplitRequestEvent) {
+      if (attemptNumber > 0 && splitIdAssignedBySubtask.containsKey(subTaskId)) {
+        LOG.info(
+            "Received request split event from subtask {} attempt {}", subTaskId, attemptNumber);
+        enumeratorContext.assignSplit(splitIdAssignedBySubtask.get(subTaskId), subTaskId);
+      } else if (attemptNumber == 0) {
```

Comment:
Why is this unexpected?

Reply:
Refactored the PR based on the comments above. This is no more applicable.